script.js

```diff
@@ -1,4 +1,262 @@
+    var currentSuit = suits[suitIndex];
+    // console.log("current suit: " + currentSuit);
+
+    // loop to create all cards in this suit: rank 1-13
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      // For special cards
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+      // console.log("rank: " + rankCounter);
+      // add the card to the deck
+      deck.push(card);
+      rankCounter = rankCounter + 1;
+    }
+    suitIndex = suitIndex + 1;
+  }
+  return deck;
+};
+
+var getRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+
+var shuffleCards = function (cards) {
+  var index = 0;
+  while (index < cards.length) {
+    var randomIndex = getRandomIndex(cards.length);
+    var currentItem = cards[index];
+    var randomItem = cards[randomIndex];
+    cards[index] = randomItem;
+    cards[randomIndex] = currentItem;
+    index = index + 1;
+  }
+  return cards;
+};
+
+var deck = shuffleCards(makeDeck());
+
+// when game ends
+var endGame = false;
+var ace = false;
+// To denote that the player should get a max sum of 21.
+var maxSumOfCards = 21;
+// If player chose to stand, then player can no longer hit until game is over.
+var playerChoseStand = false;
+// to keep track of the players' cards
+var playerHand = [];
+var dealerHand = [];
+var index2 = 1;
+others = "";
+var acesInHand = 0;
+var aceValue = 0;
+
+// Deals card to a hand - player or dealer
+var dealCardToHand = function (hand) {
+  hand.push(deck.pop());
+  return hand;
+};
+
+// Get sum of all cards in a hand
+var getSumOfHand = function (hand) {
+  var acesInHand = 0;
+  var SumOfCards = 0;
+  var index = 0;
+  while (index < hand.length) {
+    var currentCard = hand[index];
+
+    if (currentCard.rank >= 2 && currentCard.rank <= 10) {
+      // value and rank of card is the same for cards 2 to 10.
+      SumOfCards = SumOfCards + currentCard.rank;
+    } else if (currentCard.rank >= 11 && currentCard.rank <= 13) {
+      // value of J,Q,K is 10.
+      SumOfCards = SumOfCards + 10;
+    } else if (currentCard.rank == 1) {
+      acesInHand = acesInHand + 1;
+      SumOfCards = SumOfCards + 11;
+    }
+    index = index + 1;
+  }
+
+  if (SumOfCards > maxSumOfCards && acesInHand > 0) {
+    var index = 0;
+    while (index < acesInHand) {
+      SumOfCards = SumOfCards - 10;
+      if (SumOfCards <= maxSumOfCards) {
+        break;
+      }
+      index = index + 1;
+    }
+  }
+
+  return SumOfCards;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (endGame == true) {
+    endGame = true;
```

Comment:
I am not too sure if this line is necessary. If the code managed to reach this line, it means that endGame is already true